crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -1583,6 +1695,34 @@ fn mul_reg64_reg64(buf: &mut Vec<'_, u8>, src: X86_64GeneralReg) {
     buf.extend(&[rex, 0xF7, 0b1110_0000 | (src as u8 % 8)]);
 }
 
+/// `IDIV r/m64` -> Signed Divide r/m64 to r64.
+#[inline(always)]
+fn idiv_reg64_reg64(buf: &mut Vec<'_, u8>, src: X86_64GeneralReg) {
+    let mut rex = REX_W;
+    rex = add_reg_extension(src, rex);
+
+    if src.value() > 7 {
+        rex |= REX_PREFIX_B;
+    }
+
+    // adds a cqo (convert doubleword to quadword)
```

Comment:
What does this mean? Is it promoting something from 32 to 64 bit?

Reply:
I think here it turns a 64-bit value into a 128-bit one. (the `q` stands for quadword). The reason for it must be that `idiv` is really

> Signed divide RDX:RAX by r/m64, with result stored in RAX ← Quotient, RDX ← Remainder.

so you're kind of clearing the RDX register by extending the integer (is what I understand)

for unsigned division, rust generates

```
 xor    edx,edx
 div    rcx
```

again clearing edx

Reply:
Oh. Makes total sense.